### PR TITLE
allow setting max deal prices config via cli flag

### PR DESCRIPTION
**Pull request by en0ma**
Some node operators would like to set these values as they see fit

**verifications**
- I have verified the defaults are picked up correctly
- I have verified both flags when set are picked up corretcly
- I have verified no changes in code behavior for defaults
- I have verified SPs are filtered correctly when prices are set

### CODE REVIEW

**Review by jimmylee — APPROVED** [review]
LGTM, lets get another +1 before landing it though.

**Comment by alvin-reyes** [comment]
@en0ma this looks good. 

Please add the following on this PR:

- verification tests - it doesn't need to be a formal doc. Just screenshots and results of non-negative impact is ok.
- OR regression test/edge cases - can we run a few deal making test? What if the miner sets a relatively higher price?
- OR unit tests

Honestly any verification test would be good to include here.

File: constants/constants.go [hunk 1]
```diff
@@ -44,3 +45,14 @@ const TokenExpiryDurationRegister = time.Hour * 24 * 7          // 1 week
 const TokenExpiryDurationLogin = time.Hour * 24 * 30            // 30 days
 const TokenExpiryDurationDefault = time.Hour * 24 * 30          // 30 days
 const TokenExpiryDurationPermanent = time.Hour * 24 * 365 * 100 // 100 years
+
+var DealMaxPrice abi.TokenAmount
+var VerifiedDealMaxPrice = abi.NewTokenAmount(0)
+
+func init() {
```
Comment (alvin-reyes) [inline constants/constants.go:52]: Why did we need to move this init() to constants?

Comment (en0ma) [reply constants/constants.go:52]: no particular reason besides colocating default values (given all it does is set the default price)

Comment (alvin-reyes) [reply constants/constants.go:52]: Ok I see that you're trying to initialize the value of the dealmax price so this change seems reasonable

**Review by alvin-reyes — APPROVED** [review]
LGTM
